<?xml version="1.0"?>

<launch>

    <!-- start gazebo world -->
    <include file="$(find hector_gazebo_worlds)/launch/start.launch">
        <arg name="world" value="$(find aro)/world/generated.world"/>
    </include>

    <group ns="uav1">
        <!-- spawn uav1 -->
        <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
            <arg name="name" value="uav1" />
            <arg name="tf_prefix" value="uav1" />
            <arg name="model" value="$(find aro)/world/quadrotor_urdf/uav1.gazebo.xacro" />
            <arg name="x" value="8" />
            <arg name="y" value="4.0" />
            <arg name="z" value="1.1" />
        </include>
        <!-- start uav1/hector_node -->
        <node pkg="aro" type="hector_node.py" name="hector_node" output="screen" args="1">
            <rosparam command="load" file="$(find aro)/config/simulation.yaml" />
            <remap from="/release" to="/uav1/release"/>
            <remap from="/ground_truth/state" to="/uav1/ground_truth/state"/>
            <remap from="/cmd_vel" to="/uav1/cmd_vel"/>
            <!-- uav1 specific -->
            <remap from="/sonar_height" to="/uav1/sonar_height"/>
            <!-- <remap from="/scan" to="/uav1/scan"/> -->
        </node>
    </group>

    <group ns="uav2">
        <!-- spawn uav2 -->
        <include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch">
            <arg name="name" value="uav2" />
            <arg name="tf_prefix" value="uav2" />
            <arg name="model" value="$(find aro)/world/quadrotor_urdf/uav2.gazebo.xacro" />
            <arg name="x" value="8" />
            <arg name="y" value="6" />
            <arg name="z" value="1.1" />
        </include>
        <!-- start uav2/hector_node -->
        <node pkg="aro" type="hector_node.py" name="uav2_hector_node" output="screen" args="2">
            <rosparam command="load" file="$(find aro)/config/simulation.yaml" />
            <remap from="/release" to="/uav2/release"/>
            <remap from="/ground_truth/state" to="/uav2/ground_truth/state"/>
            <remap from="/cmd_vel" to="/uav2/cmd_vel"/>
            <!-- uav2 specific -->
            <remap from="/fertilizer_stock" to="/uav2/fertilizer_stock"/>
        </node>
    </group>

</launch>